pkg/source/source.go

```diff
@@ -228,7 +273,7 @@ var _ Source = &Informer{}
 
 // Start is internal and should be called only by the Controller to register an EventHandler with the Informer
```

Comment:
Should `Handler    handler.EventHandler` be handler.TypedEventHandler? (same for predicates)

(disclaimer: might not make any sense at all :))

Reply:
Additionally, I see there is a mismatch in the sense that NewEventHandler only works with client.Object while TypedEventHandler could handle any. Wondering if NewEventHandler should be able to handle any

Reply:
I've left it without generics under the assumption that this is specific to kube objects and thus will always use reconcile.Request. I guess that might not be true in same cases, like @sttts wanting to have the cluster in the request type

Reply:
I played around with this. This would be a very breaking change the the `Informer` source and there is no good way to infer the `object` type even when changing it to be initialized through a constructor func, because there is nothing to infer it from - The informer doesn't have it in its signature.

My understanding is this source only exists to be able to re-use an existing client-go informer. In 99% of cases, ppl should be using `source.Kind`.

I think it is ok not to support the typing with this source and if we change our mind on that/ppl ask for it, we can still do it later on.

Reply:
Sounds good, thank you very much for looking into it!